### PR TITLE
Create tour page

### PR DESCRIPTION
**Pull request by CDovel**
# Overview
Functionality added for creating a tour. Creating a tour contains adding name, description, location, and images for the tour. The images are added by transitional hotspot (a "location") in the tour. Each transitional hotspot is added by adding a group of images that make up that hotspot. Once all the hotspots are added the tour is sent to the back end and computed.

This page is accessed through the hamburger menu.

## Testing
<not ready>

### CODE REVIEW

File: virotour/lib/src/tour/tour_create_view.dart [hunk 1]
```diff
@@ -0,0 +1,245 @@
+// tour_edit_view.dart
```
Comment (tpbnick) [inline virotour/lib/src/tour/tour_create_view.dart]: remove

File: virotour/lib/src/tour/tour_create_view.dart [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+                      //   headers: {'Content-Type': 'application/json'},
+                      //   body: json.encode(body),
+                      // );
+                      // if (response.statusCode == 200) {
+                      //   Navigator.pop(context);
+                      // } else {
+                      //   showDialog(
+                      //     context: context,
+                      //     builder: (context) => AlertDialog(
+                      //       title: const Text('Failed to update tour'),
```
Comment (tpbnick) [inline virotour/lib/src/tour/tour_create_view.dart]: Change to "Failed to create tour"

File: virotour/lib/src/tour/tour_create_view.dart [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+                      style: TextStyle(
+                        color: Colors.white,
+                      ),
+                    ),
+                  ),
+                  const SizedBox(width: 16.0),
+                  ElevatedButton(
+                    onPressed: () async {
+                      // final url =
+                      //     'http://127.0.0.1:8081/api/update/tour/${widget.tour.id}';
```
Comment (tpbnick) [inline virotour/lib/src/tour/tour_create_view.dart]: Change to the create tour endpoint

**Comment by nguyendviet** [comment]
- [x] Merge https://github.com/umgc/spring2023/pull/33 @menchukf Can you merge yours to unblock Christian?

**Comment by tpbnick** [comment]
> * [x]  Merge [Hamburger menu #33](https://github.com/umgc/spring2023/pull/33) @menchukf Can you merge yours to unblock Christian?

Just merged.

**Comment by tpbnick** [comment]
I linked up the `tour_create_view` with the hamburger menu.  @CDovel @nguyendviet

**Comment by nguyendviet** [comment]
@CDovel Thank you for your work!

There are some issues with your code:
- [ ] It works on Chrome. I can select photos and remove them from the list. Can you make the Save button to go to the Tour page? In the future, it'll lead to the newly created tour. Please specify that in a comment. Once we click Save, it'll send all the images to the blob storage, wait for the response from the server and redirect to the Tour.
- [ ] It doesn't work on iOS due to
```
Exception has occurred.
PlatformException (PlatformException(invalid_image, Cannot load representation of type public.jpeg, NSItemProviderErrorDomain, null))
```
Error on [this line](https://github.com/umgc/spring2023/blob/acec1d2d34c3d4d1ee9c8406dfb08bf0d2cedfd0/virotour/lib/src/tour/tour_create_view.dart#L125).
Screenshot 2023-03-11 at 20 50 01
- [ ] It doesn't work on Android for a common error:
Screenshot 2023-03-11 at 20 48 27
This is because on Android, a request needs a Header. You can Google/ChatGPT a solution.

File: virotour/test/widget_test.dart [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:virotour/main.dart';
+
+void main() {
+  testWidgets('Counter increments smoke test', (WidgetTester tester) async {
+    // Build our app and trigger a frame.
+    await tester.pumpWidget(const MyApp());
```
Comment (nguyendviet) [inline virotour/test/widget_test.dart]:
```suggestion
 testWidgets('Counter increments smoke test', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    final settingsController = SettingsController(SettingsService());
    await tester.pumpWidget(App(
      settingsController: settingsController,
    ));
```

File: virotour/test/widget_test.dart [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+//
+// To perform an interaction with a widget in your test, use the WidgetTester
+// utility in the flutter_test package. For example, you can send tap and scroll
+// gestures. You can also use WidgetTester to find child widgets in the widget
+// tree, read text, and verify that the values of widget properties are correct.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:virotour/main.dart';
```
Comment (nguyendviet) [inline virotour/test/widget_test.dart]:
```suggestion
import 'package:virotour/src/app.dart';
import 'package:virotour/src/settings/settings_controller.dart';
import 'package:virotour/src/settings/settings_service.dart';
```

**Comment by CDovel** [comment]
@nguyendviet from what i've found online that problem with iOS seems to be either a corrupt image that you're trying to send or a bug with the iOS simulator. That library should be able to support iOS. The android problem I will make sure to add a header.

**Comment by CDovel** [comment]
The header should be added now to work on android.

**Review by nguyendviet — CHANGES_REQUESTED** [review]
@CDovel Thank you for your work. Your feature works great on Chrome!

- I've rebased your branch and resolved all conflicts to save you some time.
- On Android, it has a margin error. It's a small one but it'll affect our demo. Can you fix it?
Screenshot 2023-03-15 at 12 33 17

- On iOS, it throws an error. You'll need to fix this.
Screenshot 2023-03-15 at 11 59 05

**Comment by CDovel** [comment]
Hey @nguyendviet my most recent commit should fix the android problem. I still need to look closer into the issues on iOS.

**Comment by nguyendviet** [comment]
Unfortunately, it still doesn't work. Android has many screen sizes. I recommend you do something like [this](https://github.com/umgc/spring2023/blob/team_a_dev/virotour/lib/src/tour/tour_details_view.dart#L37-L42).
Screenshot 2023-03-15 at 21 58 52

**Review by nguyendviet — APPROVED** [review]
I've checked it. Great work, Chris!

**Pull request merged**